### PR TITLE
[TAS-503] Yokowaga GS200 Driver

### PR DESCRIPTION
**Pull request by fedonman**
Added Yokogawa `GS200` instrument and associated istrument controller.

### CODE REVIEW

**Review by jjmartinezQT — CHANGES_REQUESTED** [review]

File: src/qililab/instruments/yokogawa/gs200.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    def voltage(self, value: float):
+        """Set Yokogawa GS200 `voltage` property. If `ramping_enabled` is set to True, the voltage will transition linearly according to `ramping_rate`. Else, it will be set instantly."""
+        if self.ramping_enabled:
+            self.device.ramp_voltage(value, self.ramping_rate * 0.001, 0.001)
+        else:
+            self.device.voltage(value)
+        self.settings.voltage[0] = value
+
+    @Instrument.CheckDeviceInitialized
+    def setup(self, parameter: Parameter, value: float | str | bool, channel_id: int | None = None):
```
[the remark this answers is not in the document]
Comment (fedonman) [reply src/qililab/instruments/yokogawa/gs200.py:161]: It's the same signature we use in all instruments.

[the remark this answers is not in the document]
Comment (jjmartinezQT) [reply src/qililab/instruments/yokogawa/gs200.py:161]: Fair enough, dismiss this one then.

File: tests/instruments/yokogawa/test_yokogawa_gs200.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    def test_initial_setup_method(self, yokogawa_gs200: GS200):
+        """Test the initial setup method"""
+        yokogawa_gs200.initial_setup()
+
+    def test_ramping_enabled_property(self, yokogawa_gs200: GS200):
+        """Test the source mode property"""
+        assert hasattr(yokogawa_gs200, "ramping_enabled")
+        assert yokogawa_gs200.ramping_enabled == yokogawa_gs200.settings.ramping_enabled[0]
+        yokogawa_gs200.ramping_enabled = False
+        assert yokogawa_gs200.ramping_enabled is False
```
Comment (jjmartinezQT) [inline tests/instruments/yokogawa/test_yokogawa_gs200.py:104]:
```suggestion
        assert not yokogawa_gs200.ramping_enabled
```

This complies with Sourcery on style.

**Review by jjmartinezQT — CHANGES_REQUESTED** [review]

**Review by jjmartinezQT — APPROVED** [review]
Good work! Ready to go.

**Review by jjmartinezQT — CHANGES_REQUESTED** [review]
Good to go once code coverage is covering the missing line.

**Review by jjmartinezQT — APPROVED** [review]
Great work.

**Pull request merged**